IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -455,47 +386,38 @@ - (BOOL)removeItemsWithAccountKey:(MSIDCacheKey *)key
                           context:(id<MSIDRequestContext>)context
                             error:(NSError **)error
 {
-    MSID_TRACE;
-    MSID_LOG_VERBOSE(context, @"Remove keychain items, key info (account: %@ service: %@, keychainGroup: %@)", _PII_NULLIFY(key.account), _PII_NULLIFY(key.service), [self keychainGroupLoggingName]);
-    MSID_LOG_VERBOSE_PII(context, @"Remove keychain items, key info (account: %@ service: %@, keychainGroup: %@)", key.account, key.service, self.keychainGroup);
-
-    if (!key)
-    {
-        [self createError:@"Key is nil."
-                   domain:MSIDErrorDomain errorCode:MSIDErrorInvalidDeveloperParameter error:error context:context];
-        return NO;
-    }
-
-    NSMutableDictionary *query = [self primaryAccountAttributesForKey:key];
-    [query addEntriesFromDictionary:[self secondaryAccountAttributesForKey:key]];
-    query[(id)kSecMatchLimit] = (id)kSecMatchLimitAll;
-
-    MSID_LOG_INFO(context, @"Trying to delete keychain items...");
-    MSID_LOG_VERBOSE_PII(context, @"SecItemDelete: query=%@", query);
-    OSStatus status = SecItemDelete((CFDictionaryRef)query);
-    MSID_LOG_INFO(context, @"Keychain delete status: %d", (int)status);
-
-    if (status != errSecSuccess && status != errSecItemNotFound)
-    {
-        [self createError:@"Failed to remove multiple accounts from keychain"
-                   domain:MSIDKeychainErrorDomain errorCode:status error:error context:context];
-        return NO;
-    }
-
-    return YES;
+    return [self removeItemsWithKey:key context:context error:error];
```

Comment:
this is a super dangerous operation on Mac with login keychain. Do we want to add some more protection here?

Reply:
The removeItemsWithKey api is used internally by the library and we will always create the appropriate key. Can you please explain how it is dangerous. If we create the key in the right manner, I do not see this removing any other items in the login keychain.

Reply:
As discussed, added check as follows: To remove any items, either account or service attribute needs to be specified in the key.